base/floatfuncs.jl

```diff
@@ -19,11 +19,13 @@ signbit(x::Float16) = signbit(bitcast(Int16, x))
 """
     maxintfloat(T=Float64)
 
-The largest consecutive integer that is exactly represented in the given floating-point type `T`
-(which defaults to `Float64`).
+The largest consecutive integer-valued floating point number that is exactly represented in
+the given floating-point type `T` (which defaults to `Float64`).
 
-That is, `maxintfloat` returns the smallest positive integer `n` such that `n+1`
-is *not* exactly representable in the type `T`.
+That is, `maxintfloat` returns the smallest positive integer-valued floating point number
+`n` such that `n+1` is *not* exactly representable in the type `T`.
+
+When an integer is needed, use `Integer(maxintfloat(T))`.
```

Comment:
When an integer type is needed

Reply:
I am not sure I understand, as `Integer(maxintfloat(T))` will return an integer (value), not a type.

Reply:
“Integer-typed value”?

Reply:
The point is that `maxintfloat(T)` returns an integer already, so your current wording is misleading.

"When an `Integer`-type value is needed" seems okay.